cpu/cpu.go

```diff
@@ -340,23 +351,24 @@ func getStats(path string, stats map[string]interface{}, prevMetricsSum map[stri
 				}
 			}
 
-			metricStats[getNamespaceMetricPart(metricName, percentageRepresentationType)] = float64(0)
+			metricStats[getNamespaceMetricPart(metricName, percentageRepresentationType)] = nil
 
 			if mapKeyExists(cpuID, prevMetricsSum) {
 				diffSum := currDataSum - prevMetricsSum[cpuID]
-				if diffSum != 0 {
-					prevVal, err := getMapFloatValueByNamespace(stats,
-						[]string{cpuID, getNamespaceMetricPart(metricName, jiffiesRepresentationType)})
+				if diffSum > 0 {
+					prevVal, err := getMapFloatValueByNamespace(stats[cpuID],
+						[]string{getNamespaceMetricPart(metricName, jiffiesRepresentationType)})
 					if err != nil {
 						return err
 					}
 
 					if percVal := float64(100 * (currVal - prevVal) / diffSum); percVal < 0 {
```

Comment:
100 is a number of scheduler ticks per second, but that can vary between systems.
This constant should be extracted to a descriptive constant, or better - retrieved from OS.